client/data/newsletter-categories/use-unset-newsletter-category-mutation.tsx

```diff
@@ -0,0 +1,40 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import request from 'wpcom-proxy-request';
+
+type UnsetNewsletterCategoryResponse = {
+	success: boolean;
+};
+
+const useUnsetNewsletterCategoryMutation = ( siteId: string | number ) => {
+	const queryClient = useQueryClient();
+	const cacheKey = [ `newsletter-categories-${ siteId }` ];
+
+	return useMutation( {
+		mutationFn: async ( id: number ) => {
+			if ( ! id ) {
+				throw new Error( 'ID is missing.' );
+			}
+
+			const response = await request< UnsetNewsletterCategoryResponse >( {
+				path: `/sites/${ siteId }/newsletter-categories/${ id }`,
+				method: 'DELETE',
+				apiVersion: '2',
+				apiNamespace: 'wpcom/v2',
+			} );
+
+			if ( ! response.success ) {
+				throw new Error( 'Something went wrong while removing category as newsletter category.' );
+			}
+
+			return response;
+		},
+		onMutate: async () => {
+			await queryClient.cancelQueries( cacheKey );
```

Comment:
Would it already be possible to do an optimistic update here, by mutation the cache?